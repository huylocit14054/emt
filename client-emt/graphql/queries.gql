query getCurrentUser {
  currentUser {
    id
    username
    role
    avatar
  }
}

query getCurrentUserProfile {
  profile: currentUser {
    id
    username
    email
    role
    avatar
  }
}

query getCurrentUserOnClient {
  currentUser @client {
    id
    username
    role
    avatar
  }
}

query getAllUsers {
  users {
    avatar
    id
    username
    email
    createdAt
  }
}

query getProjectOfCurrentUser {
  projectsAsMemberOfCurrentUser {
    id
    name
    description
  }
  projectsAsAdminOfCurrentUser {
    id
    name
    description
  }
}

query getMembersByProjectId($projectId: ID!) {
  projectMembers(projectId: $projectId) {
    id
    role
    user {
      id
      email
      username
      role
      avatar
    }
  }
}

query getProjectById($id: ID!) {
  project(id: $id) {
    id
    name
    description
    isManagedByCurrentUser
  }
}
