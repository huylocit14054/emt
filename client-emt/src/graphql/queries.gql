#import "./fragments/BasicRuleInfo.gql"
#import "./fragments/BasicCompanyInfo.gql"
query getCurrentUser {
  currentUser {
    id
    username
    role
    avatar
  }
}

query getClientCurrentUser {
  currentUser @client {
    id
    username
    role
    avatar
  }
}

query usersSuggestion($query: String!) {
  usersSuggestion(query: $query) {
    id
    username
    email
    avatar
  }
}

query membersSuggestion($query: String!, $projectId: ID!) {
  membersSuggestion(query: $query, projectId: $projectId) {
    id
    username
    email
    avatar
  }
}

query getCurrentUserProfile {
  profile: currentUser {
    id
    username
    email
    role
    avatar
  }
}

query getCurrentUserOnClient {
  currentUser @client {
    id
    username
    role
    avatar
  }
}

query getAllUsers {
  users {
    avatar
    id
    username
    email
    createdAt
  }
}

query getProjectOfCurrentUser {
  projectsAsMemberOfCurrentUser {
    id
    name
    description
    shortenMembers {
      id
      avatar
    }
    memberCount
  }
  projectsAsAdminOfCurrentUser {
    id
    name
    description
    shortenMembers {
      id
      avatar
    }
    memberCount
  }
}

query getMembersByProjectId($projectId: ID!) {
  projectMembers(projectId: $projectId) {
    id
    role
    status
    user {
      id
      email
      username
      role
      avatar
    }
  }
}

query getAssignmentsByProjectId($projectId: ID!) {
  assignments: projectMembers(projectId: $projectId) {
    id
    user {
      id
      username
      avatar
    }
    authorizedDimensions
    authorizations {
      dimension {
        id
      }
      optionAuthorizations {
        option {
          id
          name
          dimension {
            id
          }
        }
      }
    }
  }
  projectDimensions(projectId: $projectId) {
    id
    name
    category
  }
}

query getDimensionsByProjectId($projectId: ID!) {
  projectDimensions(projectId: $projectId) {
    id
    name
    category
    createdAt
    options {
      id
      name
    }
  }
}

query getDimensionSuggestionsByProjectId($projectId: ID!) {
  projectDimensions(projectId: $projectId) {
    id
    name
  }
}

query getProjectDimensionsTreeByProjectId($projectId: ID!) {
  projectDimensionsTree(projectId: $projectId)
}

query getOptionsByDimensionId($dimensionId: ID!) {
  dimensionOptions(dimensionId: $dimensionId) {
    id
    name
  }
}

query getProjectById($id: ID!) {
  project(id: $id) {
    id
    name
    description
    isManagedByCurrentUser
  }
}

query getAssignmentsByMemberId($memberId: ID!) {
  memberAssignments(memberId: $memberId)
}

query getRulesByProjectId($projectId: ID!) {
  projectRules(projectId: $projectId) {
    ...BasicRuleInfo
  }
}

query getAssignmentsOfCurrentMember($projectId: ID!) {
  currentAppliedRule: currentAppliedRuleOfProject(projectId: $projectId) {
    id
    ruleStringToDisplay
  }
  assignments: memberAssignmentsDetails(projectId: $projectId) {
    dimension {
      id
      name
      category
    }
    optionAuthorizations {
      option {
        id
        name
      }
    }
  }
}

query currentMemberUtmHistory($projectId: ID!) {
  currentMemberUtmHistory: memberUtmHistory(projectId: $projectId) {
    id
    url
    createdAt
  }
}

query utmAnalysis($projectId: ID!) {
  utmAnalysis(projectId: $projectId) {
    id
    url
    createdAt
    projectMember {
      id
      user {
        username
        id
        avatar
      }
    }
  }
}

query getAllPlansOfApplication {
  allPlans {
    id
    name
  }
}

query getAllCompaniesOfApplication {
  allCompanies {
    ...BasicCompanyInfo
  }
}

query getCompany($companyId: ID!) {
  company(companyId: $companyId) {
    id
    logo
    name
    currentUserAsMember {
      id
      roles
    }
  }
}
