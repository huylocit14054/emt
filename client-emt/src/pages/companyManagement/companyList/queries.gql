query getCompanyDetails($companyId: ID!, $memberAmount: Int!){
    company(companyId: $companyId){
        id
        status
        name
        description
        email
        address
        phoneNumber
        plan {
            id
            name
        }
        companyMemberCount
        companyMembers(first: $memberAmount) {
            id
            roles
            user {
                id
                username
                avatar
            }
        }
    }
}
