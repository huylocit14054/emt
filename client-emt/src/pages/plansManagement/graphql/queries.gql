#import "./fragments/BasicPlanInfo.gql"

query allPlans{
  allPlans{
    ...BasicPlanInfo
  }
}

query getAllServices {
  services {
    id
    name
  }
}
