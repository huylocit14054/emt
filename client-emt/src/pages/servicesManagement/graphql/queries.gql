#import "./fragments/BasicServiceInfo.gql"
query getAllServices {
  services {
    ...BasicServiceInfo
  }
}