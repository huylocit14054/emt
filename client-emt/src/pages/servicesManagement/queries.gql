query getAllServices {
  services {
    id
    name
    createdAt
    description
  }
}