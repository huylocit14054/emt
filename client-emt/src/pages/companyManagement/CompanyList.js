import React from 'react';
import { Query } from 'react-apollo';
import { Table } from 'antd';
import TimeAgo from 'react-timeago';
import { getAllCompaniesOfApplication as GET_ALL_COMPANIES_QUERY } from '../../graphql/queries.gql';
import CloudImage from '../../components/CloudImage';
import { CompanyDetailsDrawer } from './companyList/CompanyDetailsDrawer';
import renderStatus from '../../utils/renderStatus';
import renderPlan from '../../utils/renderPlan';

const { Column } = Table;

const CompanyList = () => (
  <Query query={GET_ALL_COMPANIES_QUERY}>
    {({ loading, error, data }) => {
      if (error) return <div>Error! {error.message}</div>;
      return (
        <Table
          bordered
          pagination={{ pageSize: 8 }}
          loading={loading}
          dataSource={data.allCompanies}
          expandedRowRender={company => <p style={{ margin: 0 }}>{company.description}</p>}
          rowKey="id"
        >
          <Column
            title="Name"
            key="name"
            render={company => (
              <a>
                {' '}
                <CloudImage
                  publicId={company.logo}
                  width={40}
                  height={40}
                  style={{ marginRight: 20 }}
                >
                  {company.name}
                </CloudImage>
              </a>
            )}
          />
          <Column
            align="center"
            title="Plan"
            dataIndex="plan"
            render={plan => renderPlan(plan.name)}
          />
          <Column align="center" title="Email" dataIndex="email" />
          <Column
            align="center"
            title="Joined"
            dataIndex="createdAt"
            render={date => <TimeAgo date={date} />}
          />
          <Column
            align="center"
            title="Status"
            dataIndex="status"
            render={status => renderStatus(status)}
          />
          <Column
            align="center"
            title="Action"
            key="action"
            render={company => (
              <span>
                <CompanyDetailsDrawer
                  companyId={company.id}
                  companyName={company.name}
                  companyLogo={company.logo}
                />
              </span>
            )}
          />
        </Table>
      );
    }}
  </Query>
);

export default CompanyList;
