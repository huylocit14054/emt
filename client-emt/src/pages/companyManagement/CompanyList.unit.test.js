import React from 'react';
import { mount } from 'enzyme';
import { MockedProvider } from 'react-apollo/test-utils';
import { getAllCompaniesOfApplication as GET_ALL_COMPANIES_QUERY } from '../../graphql/queries.gql';
import wait from '../../utils/wait';
import CompanyList from './CompanyList';

describe('(Component) CompanyList', () => {
  const mocks = [
    {
      request: {
        query: GET_ALL_COMPANIES_QUERY,
      },
      result: {
        data: {
          allCompanies: [
            {
              logo: 'default-avatar_wbcfln.png',
              id: '52',
              name: 'hahahahahah',
              description: 'quat',
              status: 'active',
              plan: {
                id: '1',
                name: 'plan 1',
                __typename: 'Plan',
              },
              email: 'company@gmail.com',
              createdAt: Date.now(),
              __typename: 'Company',
            },
          ],
        },
      },
    },
  ];
  it('renders...', async () => {
    const wrapper = mount(
      <MockedProvider mocks={[]}>
        <CompanyList />
      </MockedProvider>
    );

    expect(wrapper).to.have.length(1);
  });

  it('should render loading state initially', () => {
    const wrapper = mount(
      <MockedProvider mocks={[]}>
        <CompanyList />
      </MockedProvider>
    );

    expect(
      wrapper
        .find('Table')
        .first()
        .props().loading
    ).toBe(true);
  });

  it('should render data', async () => {
    const wrapper = mount(
      <MockedProvider mocks={mocks}>
        <CompanyList />
      </MockedProvider>
    );
    await wait(0); // wait for response
    expect(wrapper.text()).toMatchSnapshot();
  });

  it('should show error UI when error appears', async () => {
    const errorMocks = [
      {
        request: {
          query: GET_ALL_COMPANIES_QUERY,
        },
        result: {
          data: null,
        },
        error: new Error('aw fuck!'),
      },
    ];

    const wrapper = mount(
      <MockedProvider mocks={errorMocks}>
        <CompanyList />
      </MockedProvider>
    );
    await wait(0); // wait for response

    expect(wrapper.text()).toContain('Error!');
  });
});
