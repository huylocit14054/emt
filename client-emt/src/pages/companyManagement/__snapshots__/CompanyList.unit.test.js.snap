// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Component) CompanyList should render data 1`] = `"NamePlanEmailJoinedStatusAction  hahahahahahplan 1company@gmail.com0 seconds agoactiveView1"`;
